src/sf-mcp-server.ts

```diff
@@ -0,0 +1,141 @@
+   * @param {ServerOptions & { telemetry?: Telemetry }} [options] - Optional server configuration including telemetry
+   */
+  public constructor(serverInfo: Implementation, options?: ServerOptions & { telemetry?: Telemetry }) {
+    super(serverInfo, options);
+    this.telemetry = options?.telemetry;
+  }
+
+  /**
+   * Registers a zero-argument tool `name`, which will run the given function when the client calls it.
+   */
+  public tool(name: string, cb: ToolCallback): RegisteredTool;
+  /**
+   * Registers a zero-argument tool `name` (with a description) which will run the given function when the client calls it.
+   */
+  public tool(name: string, description: string, cb: ToolCallback): RegisteredTool;
+  /**
+   * Registers a tool taking either a parameter schema for validation or annotations for additional metadata.
+   * This unified overload handles both `tool(name, paramsSchema, cb)` and `tool(name, annotations, cb)` cases.
+   *
+   * Note: We use a union type for the second parameter because TypeScript cannot reliably disambiguate
+   * between ToolAnnotations and ZodRawShape during overload resolution, as both are plain object types.
+   */
+  public tool<Args extends ZodRawShape>(
+    name: string,
+    paramsSchemaOrAnnotations: Args | ToolAnnotations,
+    cb: ToolCallback<Args>
+  ): RegisteredTool;
+  /**
+   * Registers a tool `name` (with a description) taking either parameter schema or annotations.
+   * This unified overload handles both `tool(name, description, paramsSchema, cb)` and
+   * `tool(name, description, annotations, cb)` cases.
+   *
+   * Note: We use a union type for the third parameter because TypeScript cannot reliably disambiguate
+   * between ToolAnnotations and ZodRawShape during overload resolution, as both are plain object types.
+   */
+  public tool<Args extends ZodRawShape>(
+    name: string,
+    description: string,
+    paramsSchemaOrAnnotations: Args | ToolAnnotations,
+    cb: ToolCallback<Args>
+  ): RegisteredTool;
+  /**
+   * Registers a tool with both parameter schema and annotations.
+   */
+  public tool<Args extends ZodRawShape>(
+    name: string,
+    paramsSchema: Args,
+    annotations: ToolAnnotations,
+    cb: ToolCallback<Args>
+  ): RegisteredTool;
+  /**
+   * Registers a tool with description, parameter schema, and annotations.
+   */
+  public tool<Args extends ZodRawShape>(
+    name: string,
+    description: string,
+    paramsSchema: Args,
+    annotations: ToolAnnotations,
+    cb: ToolCallback<Args>
+  ): RegisteredTool;
+
+  /**
+   * Registers a tool with the server and wraps its callback with telemetry tracking
+   *
+   * @param {string} name - The name of the tool to register
+   * @param {...unknown[]} rest - Additional arguments for tool registration, with the last argument being the callback
+   * @returns {RegisteredTool} The registered tool instance
+   */
+  public tool(name: string, ...rest: unknown[]): RegisteredTool {
```

Comment:
alternative to avoid copying method signatures:
https://github.com/salesforcecli/mcp/commit/01018a09cca1816ba6a00ebd6e69a0c1ffb858e0

the SDK is v1 but they add new stuff after spec updates, would be good to avoid having to keep these up-to-date manually.